File: javascript/leap/leap.js
export const isLeap = (anoAVerificar) => {
  const divisivelPor4 = anoAVerificar % 4 === 0; // ano é divisível por 4
  const divisivelPor100 = anoAVerificar % 100 === 0; // ano é divisível por 100
  const divisivelPor400 = anoAVerificar % 400 === 0; // ano é divisível por 400
  const ehBissexto = divisivelPor4 && !divisivelPor100 || divisivelPor400; // regras bissexto
  return ehBissexto;
};

File: dart/leap/lib/leap.dart
class Leap {
  bool leapYear(int anoAVerificar) {
    bool divisivelPor4 = anoAVerificar % 4 == 0;
    bool divisivelPor100 = anoAVerificar % 100 == 0;
    bool divisivelPor400 = anoAVerificar % 400 == 0;
    bool ehBissexto = divisivelPor4 && !divisivelPor100 || divisivelPor400;
    return ehBissexto;
  }
}

void main() {
  final leap = Leap();
  print(leap.leapYear(2000)); // true
  print(leap.leapYear(2001)); // false
}
